src/popup/router/pages/Retip.vue

```diff
@@ -34,69 +37,97 @@
 import deeplinkApi from '../../../mixins/deeplinkApi';
 import AmountSend from '../components/AmountSend';
 import UrlStatus from '../components/UrlStatus';
 import Button from '../components/Button';
+import BalanceInfo from '../components/BalanceInfo';
 
 export default {
   mixins: [deeplinkApi],
-  components: { AmountSend, UrlStatus, Button },
+  components: { AmountSend, UrlStatus, Button, BalanceInfo },
   data: () => ({
     tip: {},
     amount: null,
     amountError: false,
     loading: false,
     minCallFee: null,
+    tippingContract: null,
   }),
   computed: {
     ...mapGetters(['account', 'tippingSupported', 'minTipAmount', 'activeNetwork']),
-    ...mapState(['balance', 'tippingV1', 'sdk']),
+    ...mapState('fungibleTokens', ['selectedToken']),
+    ...mapState(['balance', 'tippingV1', 'tippingV2', 'sdk']),
     urlStatus() {
       return this.$store.getters['tipUrl/status'](this.tip.url);
     },
+    shouldUseV2Contract() {
+      return this.$route.query.id.includes('_v2') || this.$route.query.id.includes('_v3');
+    },
   },
   watch: {
     amount() {
-      this.amountError = !+this.amount || this.amount < this.minTipAmount;
+      this.amountError = !+this.amount || (!this.selectedToken && this.amount < this.minTipAmount);
     },
   },
   async created() {
     this.loading = true;
     await this.$watchUntilTruly(() => this.sdk);
     this.minCallFee = calculateFee(TX_TYPES.contractCall, {
       ...this.sdk.Ae.defaults,
-      contractId: this.activeNetwork.tipContractV1,
+      contractId: this.shouldUseV2Contract
+        ? this.activeNetwork.tipContractV2
+        : this.activeNetwork.tipContractV1,
       callerId: this.account.publicKey,
     }).min;
-    await this.$watchUntilTruly(() => this.tippingV1);
-    const tipId = +this.$route.query.id;
+    await this.$watchUntilTruly(() => this.tippingV1, this.tippingV2);
+    const tipId = this.$route.query.id;
     if (!tipId) throw new Error('"id" param is missed');
-    const { decodedResult } = await this.tippingV1.methods.get_state();
-    this.tip = tipping.getTipsRetips(decodedResult).tips.find(({ id }) => id === tipId);
+    this.tippingContract = this.shouldUseV2Contract ? this.tippingV2 : this.tippingV1;
+    const state = await this.tippingContract.methods.get_state();
+    const tipsAndRetips = await tipping.getTipsRetips(state);
+    this.tip = tipsAndRetips.tips.find(({ id }) => id === tipId);
     this.loading = false;
   },
   methods: {
     async sendTip() {
       const calculatedMaxValue =
         this.balance > this.minCallFee ? this.balance - this.minCallFee : 0;
       this.amountError = !this.amount || !this.minCallFee || calculatedMaxValue - this.amount <= 0;
-      this.amountError = this.amountError || !+this.amount || this.amount < this.minTipAmount;
+      this.amountError =
+        this.amountError ||
+        !+this.amount ||
+        (!this.selectedToken && this.amount < this.minTipAmount);
```

Comment:
This is the third time this expression appears in the same file

Reply:
Same as the above comment.